CDCgov/prime-reportstream
# Renamed /api/messages/search and update API documentation

This PR ...

- Updates the messages API documentation and code with clear and accurate descriptions
- Renames `/api/messages/search` to `/api/messages` to follow better naming practices

Test Steps:
1. Enable message tracker feature flag and confirm the renamed endpoint works. To enable the message tracker feature, go admin -> feature flags and add "message-tracker". Then, go admin -> message id search

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #8031
